open_rarity/rarity_ranker.py

```diff
@@ -1,14 +1,21 @@
 import math
+from open_rarity.models.collection import Collection
+from open_rarity.models.token_rarity import TokenRarity
+
+from open_rarity.scoring.scorer import Scorer
 
 
 class RarityRanker:
     """This class is used to rank a set of tokens given their rarity scores."""
 
+    default_scorer = Scorer()
+
     @staticmethod
-    def rank_tokens(*, token_id_to_scores: dict[str, float]) -> dict[str, int]:
-        """Ranks a set of tokens, identified uniquely by a string, based on
-        the scores provided. The ranking is done assuming @token_id_to_scores
-        contains all tokens for a given cohort/collection to rank tokens for.
+    def rank_collection(
+        collection: Collection, scorer: Scorer = default_scorer
```

Comment:
Is it necessary to assign `default_scorer=Scorer()` before passing it as the default argument? Why not just do `scorer:  Scorer = Scorer()`

Reply:
In case of reuse of the ranker class will i instantiate the Scorer one more time - the main idea to avoid the object init overhead.

Reply:
besided PEP complains if we just instantiate the class in the function call:
```
open_rarity/rarity_ranker.py:13:50: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
```